src/DataStructures/Blaze/IntegerPow.hpp

```diff
@@ -0,0 +1,75 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <blaze/math/DenseVector.h>
+#include <blaze/math/constraints/SIMDPack.h>
+#include <blaze/math/simd/BasicTypes.h>
+#include <blaze/system/Inline.h>
+#include <blaze/system/Vectorization.h>
+
+#include "Utilities/Math.hpp"
+
+namespace blaze {
+// This vectorized implementation of the integer pow function is necessary
+// because blaze does not offer its own version of a integer pow function.
+template <typename T>
+BLAZE_ALWAYS_INLINE SIMDdouble integer_pow(const SIMDf64<T>& v, const int e)
+#if BLAZE_AVX512F_MODE || BLAZE_MIC_MODE
+{
+  return _mm512_set_pd(::integer_pow((*v).eval().value[7], e),
+                       ::integer_pow((*v).eval().value[6], e),
+                       ::integer_pow((*v).eval().value[5], e),
+                       ::integer_pow((*v).eval().value[4], e),
+                       ::integer_pow((*v).eval().value[3], e),
+                       ::integer_pow((*v).eval().value[2], e),
+                       ::integer_pow((*v).eval().value[1], e),
+                       ::integer_pow((*v).eval().value[0], e));
+}
+#elif BLAZE_AVX_MODE
+{
+  return _mm256_set_pd(::integer_pow((*v).eval().value[3], e),
+                       ::integer_pow((*v).eval().value[2], e),
+                       ::integer_pow((*v).eval().value[1], e),
+                       ::integer_pow((*v).eval().value[0], e));
+}
+#elif BLAZE_SSE2_MODE
+{
+  return _mm_set_pd(::integer_pow((*v).eval().value[1], e),
+                    ::integer_pow((*v).eval().value[0], e));
+}
+#else
+{
+  return SIMDdouble{::integer_pow((*v).value, e)};
+}
+#endif
+
+struct IntegerPow {
+  int exponent;
+  explicit inline IntegerPow(const int e) : exponent(e) {}
+
+  template <typename T>
+  BLAZE_ALWAYS_INLINE decltype(auto) operator()(const T& a) const {
+    return ::integer_pow(a, exponent);
```

Comment:
I've forgotten the blaze stuff specifically for this, but this is always going to be a `double`? That is, `std::is_same_v<T, double>` is always true? If so, maybe just hard-code the `double`? If not, I would do:
```cpp
using ::integer_pow;
    return ::integer_pow(a, exponent);
```
then it'll always pick the right one.